src/sources/eventstoredb/types.rs

```diff
@@ -0,0 +1,345 @@
+use crate::event::metric::{MetricData, MetricName, MetricSeries};
+use crate::event::{Metric, MetricKind, MetricValue};
+use crate::internal_events::InternalEvent;
+use hyper::client::HttpConnector;
+use hyper::Body;
+use hyper_openssl::HttpsConnector;
+use metrics::gauge;
+use serde::de::{MapAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::collections::BTreeMap;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Stats {
+    pub proc: Proc,
+    pub sys: Sys,
+}
+
+impl Stats {
+    pub fn metrics(&self, namespace: Option<String>) -> Vec<Metric> {
+        let mut result = Vec::new();
+        let mut tags = BTreeMap::new();
+        let namespace = namespace.unwrap_or_else(|| "eventstoredb".to_string());
+
+        tags.insert("id".to_string(), self.proc.id.to_string());
+
+        let name = MetricName {
+            name: "memory_usage".to_string(),
+            namespace: Some(namespace.clone()),
+        };
+
+        let data = MetricData {
+            timestamp: None,
```

Comment:
Considering that the source it self is collecting the metrics, it should also set all timestamps to Utc::now.

Reply:
Done